packages/roc-package-web-app-react/app/server/router.js

```diff
@@ -33,7 +34,10 @@ export default function routes({ createRoutes, createStore, stats, dist }) {
                 this.status = 200;
                 this.body = renderPage();
             } else {
-                const store = createStore ? createStore() : null;
+                const { store, history, url } = setupForRender(createStore, this.url);
+
+                // Give Koa middlewares a chance to interact with the reduxStore
+                // This can be used to dynamically pass some data to the client.
```

Comment:
If we do this Koa middlewares will not have access to the `store`.

I suggest that we instead make it possible to define middlewares that run before the React renderer and after. The ones that run after then can use the `store` and the ones before can set `this.body` that we can use for an additional check in the React renderer.